classes/Hook/HookActionValidateOrder.php

```diff
@@ -53,8 +53,9 @@ public function __construct(Ps_Googleanalytics $module, Context $context)
      */
     public function run()
     {
-        // Check if we are creating backoffice order
-        if ($this->context->controller->controller_name != 'AdminOrders' && $this->context->controller->controller_name != 'Admin') {
+        // Check if we are creating backoffice order, we are only launching this hook when creating backoffice order
+        // For FO purposes, we use displayOrderConfirmation.
+        if (empty($this->context->controller->controller_name) || ($this->context->controller->controller_name != 'AdminOrders' && $this->context->controller->controller_name != 'Admin')) {
```

Comment:
```suggestion
        if (empty($this->context->controller) || ($this->context->controller->controller_name != 'AdminOrders' && $this->context->controller->controller_name != 'Admin')) {
```

Reply:
The bug in behat is because `Content::controller` is not set it is null so accessible `null->controller_name` will still fail with this fix no? 🤔

Reply:
I had a doubt to but I think `empty()` is very forgiving, `empty` will evaluate to true at the first unset property

Reply:
@jolelievre The issue was in controller_name, the fix works correctly.